project.clj

```diff
@@ -1,11 +1,15 @@
-(defproject walmartlabs/vizdeps "0.1.7-SNAPSHOT"
+(defproject clj-commons/vizdeps "0.1.7-SNAPSHOT"
   :description "Visualize Leiningen project dependencies using Graphviz."
   :url "https://github.com/walmartlabs/vizdeps"
   :license {:name "Apache Sofware License 2.0"
             :url  "http://www.apache.org/licenses/LICENSE-2.0.html"}
-  :dependencies [[org.clojure/clojure "1.8.0"]
-                 [org.clojure/tools.cli "0.3.5"]
-                 [dorothy "0.0.6"]
-                 [medley "0.8.4"]
-                 [com.stuartsierra/dependency "0.2.0"]]
-  :eval-in-leiningen true)
+  :dependencies [[org.clojure/clojure "1.9.0"]
+                 [org.clojure/tools.cli "0.4.1"]
+                 [com.stuartsierra/dependency "0.2.0"]
+                 [dorothy "0.0.7"]
+                 [medley "1.0.0"]]
+
+  :profiles {:leiningen {:dependencies [[leiningen-core "2.8.2"]
+                                        [org.apache.maven.wagon/wagon-http "3.0.0"]]}
+             :tools.deps {:dependencies [[org.clojure/tools.deps.alpha "0.5.460"]]}}
+  :aliases {"vizdeps" ["with-profile" "leiningen" "run" "-m" "clj-commons.leiningen.vizdeps/leiningen"]})
```

Comment:
> Move forward with alias pathway. Forcing an alias is only minimally more work than using a plugin and allows freedom to upgrade to newer versions of clojure.

I'm guessing that's what this is? So for anyone wanting to use vizdeps, they would need a similar alias on their project? What would it look like for them if they are adding a dependency on a deployed version? I don't think they would be able to use the `with-profile leiningen` part there? Can you explain more about how it gives freedom to upgrade to newer versions of Clojure?

Reply:
Why does the dorothy that requires an upgrade?  Where is this Dorothy 0.0.7 coming from?  Could we reasonably keep vizdeps as a plugin if we are careful about dependencies?

Reply:
Potentially for a time.  This definitely be premature - but consider this.

Clojure will be at 0.10 soon or could be considered now.  It is very difficult to figure out why a plugin doesn't work - you don't have control over your environment fundamentally.

This directions allows:
1.  Multiple system support; leiningen, tools.deps, boot, etc.
2.  Easier pathway to avoiding odd version conflicts.

The only 'requirement' I think is that it work with newer (9+)  JVMs.  As long as that is true then you could keep the plugin.

tools.deps support could also be considered premature.  

It seems to me, however, that the best move forward is to support multiple build systems and multiple jvms.  Does this change how important it is to keep the system a lein plugin?